libGeoIP/GeoIP.c

```diff
@@ -2764,3 +2764,9 @@ int GeoIP_cleanup(void) {
 
     return result;
 }
+
+void GeoIP_free(void *ptr) {
```

Comment:
free(NULL) is a valid operation. Don’t need to check for NULL.

Reply:
I don't think we need to do an autoconf check here. Our other `*_delete` functions do not check. If someone is supporting a system that doesn't follow the C standard, they can always check before they call the new function.

Reply:
> The C standard says a call free (NULL) does nothing, but some old systems don't support this (e.g., NextStep).

Pretty sure that anything not conforming to C89 has long since died a horrible death...

Reply:
> Pretty sure that anything not conforming to C89 has long since died a horrible death...

Yeah.  I was probably [trying to be a caveman](http://cvsweb.openbsd.org/cgi-bin/cvsweb/src/usr.bin/mg/Attic/theo.c?rev=1.152&content-type=text/x-cvsweb-markup) :-).